### DISCUSSION TITLE
.env with typescript

### DISCUSSION THREAD

**Discussion by visakadivines** [vercel/next.js#13328] [Help] [answered]
I'm using nextjs with typescript and I can't figure out how to get .env to work with it. When I run 'npm run dev' I see an info message saying "Loaded env from .env" however, all of my environment variables are undefined in my component files and pages. I feel like I'm missing something simple here. Any help would be much appreciated.

**Comment by sondh5** [comment, marked as answer]
Have you tried with prefix `NEXT_PUBLIC_`
https://nextjs.org/blog/next-9-4#new-environment-variables-support

Otherwise
Did you set env in `next.config.js`
```js
module.exports = {
    env: {
        ENV_VAR_NAME: process.env.ENV_VAR_NAME
    }
}
```

When you run `npm run dev`, did you see like this
```
ready - started server on http://localhost:3000
info  - Loaded env from .env
event - compiled successfully
```

**Reply by visakadivines** [reply]
Prefixing with NEXT_PUBLIC_ worked. Thank you!
